packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,739 @@
+	dataType: TestPrimitives,
+	provider: ITestTreeProvider,
+	random: IRandom,
+): Promise<void> {
+	const path: UpPath = {
+		parent: undefined,
+		parentField: rootFieldKeySymbol,
+		parentIndex: 0,
+	};
+	for (let j = 0; j < numberOfNodes; j++) {
+		const personData = generateTreeData(dataType, random);
+		tree.runTransaction((forest, editor) => {
+			const writeCursor = singleTextCursor(personData);
+			const field = editor.sequenceField(path, localFieldKey);
+			field.insert(j, writeCursor);
+			return TransactionResult.Apply;
+		});
+	}
+	await provider.ensureSynchronized();
+}
+
+async function generateEditableTree(
+	numberOfNodes: number,
+	shape: TreeShape,
+	random: IRandom,
+	dataType: TestPrimitives,
+): Promise<ISharedTree> {
+	const [provider, trees] = await createSharedTrees(
+		getTestSchema(FieldKinds.sequence),
+		[{ type: rootSchemaName }],
+		1,
+	);
+	const tree = trees[0].root;
+	assert(isUnwrappedNode(tree));
+	let field_0;
+	let currentNode;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree[createField](localFieldKey, [
+				singleTextCursor({
+					type: dataSchema.name,
+					value: generateTreeData(dataType, random),
+				}),
+			]);
+			assert(isUnwrappedNode(trees[0].root));
+			field_0 = tree[getField](localFieldKey);
+			assert(field_0 !== undefined);
+			currentNode = field_0.getNode(0);
+			for (let i = 0; i < numberOfNodes; i++) {
+				const treeData = generateTreeData(dataType, random);
+				currentNode[createField](localFieldKey, singleTextCursor(treeData));
+				currentNode = currentNode[getField](localFieldKey).getNode(0);
+			}
+			break;
+		case TreeShape.Wide:
+			assert(isUnwrappedNode(trees[0].root));
+			for (let i = 0; i < numberOfNodes - 1; i++) {
+				trees[0].root[getField](localFieldKey).insertNodes(
+					i,
+					singleTextCursor(generateTreeData(dataType, random)),
+				);
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	return trees[0];
+}
+const booleans = [true, false];
+
+function generateTreeData(dataType: TestPrimitives, random: IRandom): JsonableTree {
+	const insertValue = random.integer(Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER);
+	let map;
+	switch (dataType) {
+		case TestPrimitives.Number:
+			return { value: insertValue, type: dataSchema.name };
+		case TestPrimitives.Float:
+			return { value: insertValue, type: dataSchema.name };
+		case TestPrimitives.String:
+			return {
+				value: random.real(0, Number.MAX_SAFE_INTEGER).toString(),
+				type: dataSchema.name,
+			};
+		case TestPrimitives.Boolean:
+			return { value: random.pick(booleans), type: dataSchema.name };
+		case TestPrimitives.Map:
+			map = {
+				mapField2: {
+					mapField3: [{ type: dataSchema.name, value: insertValue.toString() }],
+				},
+			};
+			return {
+				value: map,
+				type: dataSchema.name,
+			};
+		default:
+			unreachableCase(dataType);
+	}
+}
+
+function getTestTreeAsJSObject(
+	numberOfNodes: number,
+	shape: TreeShape,
+	dataType: TestPrimitives,
+): any {
```

Comment:
It's hard to know what this function does. A more specific return type would be a great way to clarify that.

Reply:
updated return type